experimental/instrumentation-fetch/src/instrumentation.ts

```diff
@@ -0,0 +1,84 @@
+import { BaseInstrumentation, faro, globalObject } from '@grafana/faro-core';
+
+import { fetchGlobalObjectKey, originalFetch, originalFetchGlobalObjectKey } from './constants';
+
+export class FetchInstrumentation extends BaseInstrumentation {
+  readonly name = '@grafana/faro-web-sdk:instrumentation-fetch';
+  readonly version = '1.0.0'; // TODO - pull from package.json
+
+  initialize(): void {
+    console.log('Initializing FetchInstrumentation');
+    this.enable();
+  }
+
+  constructor() {
+    super();
+  }
+
+  /**
+   * Instrument fetch with Faro
+   */
+  private _instrumentFetch(): typeof fetch {
+    return function instrumentedFetch(input, init): Promise<Response> {
+      let copyInit = init ? Object.assign({}, init) : {};
+
+      let body;
+      if (copyInit && copyInit.body) {
+        body = copyInit.body;
+        copyInit.body = undefined;
+      }
+
+      const request = new Request(input, copyInit);
+      if (body) {
+        copyInit.body = body;
+      }
+
+      return originalFetch(input instanceof Request ? request : input, copyInit).then(function (response: Response) {
+        const simplifiedHeaders = {} as Record<string, string>;
+        new Headers(response.headers).forEach((v, k) => (simplifiedHeaders[`response_header_${k}`] = v));
+
+        // TODO - add instrumented functions to capture onError and onSuccess
+        faro.api.pushEvent('Response headers', simplifiedHeaders);
+        return response;
+      });
+    };
+  }
+
+  /**
+   * implements enable function
+   */
+  enable(): void {
+    Object.defineProperty(globalObject, fetchGlobalObjectKey, {
+      configurable: true,
+      enumerable: false,
+      writable: false,
+      value: this._instrumentFetch(),
+    });
+
+    Object.defineProperty(globalObject, originalFetchGlobalObjectKey, {
+      configurable: true,
+      enumerable: false,
+      writable: false,
+      value: originalFetch,
+    });
+  }
+
+  /**
+   * implements disable function
+   */
+  disable(): void {
```

Comment:
Q: why do we have a enable/didable function? In what cases do we call `disable()` and who's calling it?
Asking because the other instruments don't have that functionality.
Hint: `pause()` is handled at the `transports` level, no additional wotk for instrumentation developers :)

Reply:
Good points. I have been following the OTel fetch instrumentation and they have the enable/disable functionality. I thought it was potentially useful but hadn't really considered the use cases. Knowing that we have pause should be sufficient :+1: